// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CloudWorkloadSecurityAgentRuleCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudWorkloadSecurityAgentRuleCreateAttributes = void 0;
/**
 * Create a new Cloud Workload Security Agent rule.
 */
class CloudWorkloadSecurityAgentRuleCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CloudWorkloadSecurityAgentRuleCreateAttributes.attributeTypeMap;
    }
}
exports.CloudWorkloadSecurityAgentRuleCreateAttributes = CloudWorkloadSecurityAgentRuleCreateAttributes;
/**
 * @ignore
 */
CloudWorkloadSecurityAgentRuleCreateAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    expression: {
        baseName: "expression",
        type: "string",
        required: true,
    },
    filters: {
        baseName: "filters",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
