// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentAttachmentLinkAttributesAttachmentObject.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentAttachmentLinkAttributesAttachmentObject = void 0;
/**
 * The link attachment.
 */
class IncidentAttachmentLinkAttributesAttachmentObject {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentAttachmentLinkAttributesAttachmentObject.attributeTypeMap;
    }
}
exports.IncidentAttachmentLinkAttributesAttachmentObject = IncidentAttachmentLinkAttributesAttachmentObject;
/**
 * @ignore
 */
IncidentAttachmentLinkAttributesAttachmentObject.attributeTypeMap = {
    documentUrl: {
        baseName: "documentUrl",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
