// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentAttachmentsPostmortemAttributesAttachmentObject.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentAttachmentsPostmortemAttributesAttachmentObject = void 0;
/**
 * The postmortem attachment.
 */
class IncidentAttachmentsPostmortemAttributesAttachmentObject {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentAttachmentsPostmortemAttributesAttachmentObject.attributeTypeMap;
    }
}
exports.IncidentAttachmentsPostmortemAttributesAttachmentObject = IncidentAttachmentsPostmortemAttributesAttachmentObject;
/**
 * @ignore
 */
IncidentAttachmentsPostmortemAttributesAttachmentObject.attributeTypeMap = {
    documentUrl: {
        baseName: "documentUrl",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
