// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentAttachmentData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentAttachmentData = void 0;
/**
 * A single incident attachment.
 */
class IncidentAttachmentData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentAttachmentData.attributeTypeMap;
    }
}
exports.IncidentAttachmentData = IncidentAttachmentData;
/**
 * @ignore
 */
IncidentAttachmentData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "IncidentAttachmentAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "IncidentAttachmentRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "IncidentAttachmentType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
