// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentAttachmentLinkAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentAttachmentLinkAttributes = void 0;
/**
 * The attributes object for a link attachment.
 */
class IncidentAttachmentLinkAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentAttachmentLinkAttributes.attributeTypeMap;
    }
}
exports.IncidentAttachmentLinkAttributes = IncidentAttachmentLinkAttributes;
/**
 * @ignore
 */
IncidentAttachmentLinkAttributes.attributeTypeMap = {
    attachment: {
        baseName: "attachment",
        type: "IncidentAttachmentLinkAttributesAttachmentObject",
        required: true,
    },
    attachmentType: {
        baseName: "attachment_type",
        type: "IncidentAttachmentLinkAttachmentType",
        required: true,
    },
    modified: {
        baseName: "modified",
        type: "Date",
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
