// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LayerAttributesRestrictionsItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LayerAttributesRestrictionsItems = void 0;
/**
 * Represents a time restriction within a layer, specifying the days and times
 * when this layer is active or inactive.
 */
class LayerAttributesRestrictionsItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LayerAttributesRestrictionsItems.attributeTypeMap;
    }
}
exports.LayerAttributesRestrictionsItems = LayerAttributesRestrictionsItems;
/**
 * @ignore
 */
LayerAttributesRestrictionsItems.attributeTypeMap = {
    endDay: {
        baseName: "end_day",
        type: "LayerAttributesRestrictionsItemsEndDay",
    },
    endTime: {
        baseName: "end_time",
        type: "string",
    },
    startDay: {
        baseName: "start_day",
        type: "LayerAttributesRestrictionsItemsStartDay",
    },
    startTime: {
        baseName: "start_time",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
