// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineConfig = void 0;
/**
 * Specifies the pipeline's configuration, including its sources, processors, and destinations.
 */
class ObservabilityPipelineConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineConfig.attributeTypeMap;
    }
}
exports.ObservabilityPipelineConfig = ObservabilityPipelineConfig;
/**
 * @ignore
 */
ObservabilityPipelineConfig.attributeTypeMap = {
    destinations: {
        baseName: "destinations",
        type: "Array<ObservabilityPipelineConfigDestinationItem>",
        required: true,
    },
    processors: {
        baseName: "processors",
        type: "Array<ObservabilityPipelineConfigProcessorItem>",
        required: true,
    },
    sources: {
        baseName: "sources",
        type: "Array<ObservabilityPipelineConfigSourceItem>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
