// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineDatadogLogsDestination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineDatadogLogsDestination = void 0;
/**
 * The `datadog_logs` destination forwards logs to Datadog Log Management.
 */
class ObservabilityPipelineDatadogLogsDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineDatadogLogsDestination.attributeTypeMap;
    }
}
exports.ObservabilityPipelineDatadogLogsDestination = ObservabilityPipelineDatadogLogsDestination;
/**
 * @ignore
 */
ObservabilityPipelineDatadogLogsDestination.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    inputs: {
        baseName: "inputs",
        type: "Array<string>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineDatadogLogsDestinationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
