// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineKafkaSourceLibrdkafkaOption.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineKafkaSourceLibrdkafkaOption = void 0;
/**
 * Represents a key-value pair used to configure low-level `librdkafka` client options for Kafka sources, such as timeouts, buffer sizes, and security settings.
 */
class ObservabilityPipelineKafkaSourceLibrdkafkaOption {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineKafkaSourceLibrdkafkaOption.attributeTypeMap;
    }
}
exports.ObservabilityPipelineKafkaSourceLibrdkafkaOption = ObservabilityPipelineKafkaSourceLibrdkafkaOption;
/**
 * @ignore
 */
ObservabilityPipelineKafkaSourceLibrdkafkaOption.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    value: {
        baseName: "value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
