// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RUMApplicationCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RUMApplicationCreateAttributes = void 0;
/**
 * RUM application creation attributes.
 */
class RUMApplicationCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RUMApplicationCreateAttributes.attributeTypeMap;
    }
}
exports.RUMApplicationCreateAttributes = RUMApplicationCreateAttributes;
/**
 * @ignore
 */
RUMApplicationCreateAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
