// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleCreateRequestDataRelationshipsTeamsDataItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleCreateRequestDataRelationshipsTeamsDataItems = void 0;
/**
 * Holds the relationship data linking this schedule to a particular team,
 * identified by `id` and `type`.
 */
class ScheduleCreateRequestDataRelationshipsTeamsDataItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleCreateRequestDataRelationshipsTeamsDataItems.attributeTypeMap;
    }
}
exports.ScheduleCreateRequestDataRelationshipsTeamsDataItems = ScheduleCreateRequestDataRelationshipsTeamsDataItems;
/**
 * @ignore
 */
ScheduleCreateRequestDataRelationshipsTeamsDataItems.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "ScheduleCreateRequestDataRelationshipsTeamsDataItemsType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
