// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleUpdateRequestDataAttributesLayersItemsMembersItemsUser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleUpdateRequestDataAttributesLayersItemsMembersItemsUser = void 0;
/**
 * Identifies the user who is assigned to this member object. Only `id` is required.
 */
class ScheduleUpdateRequestDataAttributesLayersItemsMembersItemsUser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleUpdateRequestDataAttributesLayersItemsMembersItemsUser.attributeTypeMap;
    }
}
exports.ScheduleUpdateRequestDataAttributesLayersItemsMembersItemsUser = ScheduleUpdateRequestDataAttributesLayersItemsMembersItemsUser;
/**
 * @ignore
 */
ScheduleUpdateRequestDataAttributesLayersItemsMembersItemsUser.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
