// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItems = void 0;
/**
 * Defines a time restriction for a schedule layer, including which day of the week
 * it starts and ends, along with start/end times.
 */
class ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItems.attributeTypeMap;
    }
}
exports.ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItems = ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItems;
/**
 * @ignore
 */
ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItems.attributeTypeMap = {
    endDay: {
        baseName: "end_day",
        type: "ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItemsEndDay",
    },
    endTime: {
        baseName: "end_time",
        type: "string",
    },
    startDay: {
        baseName: "start_day",
        type: "ScheduleCreateRequestDataAttributesLayersItemsRestrictionsItemsStartDay",
    },
    startTime: {
        baseName: "start_time",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
