// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems = void 0;
/**
 * Defines a time restriction object for a layer within a schedule update, including
 * start and end days, as well as times.
 */
class ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems.attributeTypeMap;
    }
}
exports.ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems = ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems;
/**
 * @ignore
 */
ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems.attributeTypeMap = {
    endDay: {
        baseName: "end_day",
        type: "ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItemsEndDay",
    },
    endTime: {
        baseName: "end_time",
        type: "string",
    },
    startDay: {
        baseName: "start_day",
        type: "ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItemsStartDay",
    },
    startTime: {
        baseName: "start_time",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
