// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleCreateRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleCreateRequestDataAttributes = void 0;
/**
 * Describes the main attributes for creating a new schedule, including name, layers, time zone, and tags.
 */
class ScheduleCreateRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleCreateRequestDataAttributes.attributeTypeMap;
    }
}
exports.ScheduleCreateRequestDataAttributes = ScheduleCreateRequestDataAttributes;
/**
 * @ignore
 */
ScheduleCreateRequestDataAttributes.attributeTypeMap = {
    layers: {
        baseName: "layers",
        type: "Array<ScheduleCreateRequestDataAttributesLayersItems>",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    timeZone: {
        baseName: "time_zone",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
