// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleUserAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleUserAttributes = void 0;
/**
 * Provides basic user information for a schedule, including a name and email address.
 */
class ScheduleUserAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleUserAttributes.attributeTypeMap;
    }
}
exports.ScheduleUserAttributes = ScheduleUserAttributes;
/**
 * @ignore
 */
ScheduleUserAttributes.attributeTypeMap = {
    email: {
        baseName: "email",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
