// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleUpdateRequestDataAttributesLayersItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleUpdateRequestDataAttributesLayersItems = void 0;
/**
 * Represents a layer within a schedule update, including rotation details, members,
 * and optional restrictions.
 */
class ScheduleUpdateRequestDataAttributesLayersItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleUpdateRequestDataAttributesLayersItems.attributeTypeMap;
    }
}
exports.ScheduleUpdateRequestDataAttributesLayersItems = ScheduleUpdateRequestDataAttributesLayersItems;
/**
 * @ignore
 */
ScheduleUpdateRequestDataAttributesLayersItems.attributeTypeMap = {
    effectiveDate: {
        baseName: "effective_date",
        type: "Date",
        format: "date-time",
    },
    endDate: {
        baseName: "end_date",
        type: "Date",
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    interval: {
        baseName: "interval",
        type: "ScheduleUpdateRequestDataAttributesLayersItemsInterval",
    },
    members: {
        baseName: "members",
        type: "Array<ScheduleUpdateRequestDataAttributesLayersItemsMembersItems>",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    restrictions: {
        baseName: "restrictions",
        type: "Array<ScheduleUpdateRequestDataAttributesLayersItemsRestrictionsItems>",
    },
    rotationStart: {
        baseName: "rotation_start",
        type: "Date",
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
