// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleCreateRequestDataAttributesLayersItemsInterval.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleCreateRequestDataAttributesLayersItemsInterval = void 0;
/**
 * Defines how frequently the rotation repeats, using days and/or seconds (up to certain limits).
 */
class ScheduleCreateRequestDataAttributesLayersItemsInterval {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleCreateRequestDataAttributesLayersItemsInterval.attributeTypeMap;
    }
}
exports.ScheduleCreateRequestDataAttributesLayersItemsInterval = ScheduleCreateRequestDataAttributesLayersItemsInterval;
/**
 * @ignore
 */
ScheduleCreateRequestDataAttributesLayersItemsInterval.attributeTypeMap = {
    days: {
        baseName: "days",
        type: "number",
        format: "int32",
    },
    seconds: {
        baseName: "seconds",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
