// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringRuleCaseActionOptions.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringRuleCaseActionOptions = void 0;
/**
 * Options for the rule action
 */
class SecurityMonitoringRuleCaseActionOptions {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringRuleCaseActionOptions.attributeTypeMap;
    }
}
exports.SecurityMonitoringRuleCaseActionOptions = SecurityMonitoringRuleCaseActionOptions;
/**
 * @ignore
 */
SecurityMonitoringRuleCaseActionOptions.attributeTypeMap = {
    duration: {
        baseName: "duration",
        type: "number",
        format: "int64",
    },
    userBehaviorName: {
        baseName: "userBehaviorName",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
