// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringRuleCase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringRuleCase = void 0;
/**
 * Case when signal is generated.
 */
class SecurityMonitoringRuleCase {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringRuleCase.attributeTypeMap;
    }
}
exports.SecurityMonitoringRuleCase = SecurityMonitoringRuleCase;
/**
 * @ignore
 */
SecurityMonitoringRuleCase.attributeTypeMap = {
    actions: {
        baseName: "actions",
        type: "Array<SecurityMonitoringRuleCaseAction>",
    },
    condition: {
        baseName: "condition",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    notifications: {
        baseName: "notifications",
        type: "Array<string>",
    },
    status: {
        baseName: "status",
        type: "SecurityMonitoringRuleSeverity",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
