// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsAssertionXPathTargetTarget.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsAssertionXPathTargetTarget = void 0;
/**
 * Composed target for `validatesXPath` operator.
 */
class SyntheticsAssertionXPathTargetTarget {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsAssertionXPathTargetTarget.attributeTypeMap;
    }
}
exports.SyntheticsAssertionXPathTargetTarget = SyntheticsAssertionXPathTargetTarget;
/**
 * @ignore
 */
SyntheticsAssertionXPathTargetTarget.attributeTypeMap = {
    operator: {
        baseName: "operator",
        type: "string",
    },
    targetValue: {
        baseName: "targetValue",
        type: "any",
    },
    xPath: {
        baseName: "xPath",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "any",
    },
};
